src/coord/src/session/vars.rs

```diff
@@ -676,3 +704,96 @@ impl Value for [&str] {
         self.join(", ")
     }
 }
+
+/// Severity levels can used to be used to filter which messages get sent
+/// to a client.
+///
+/// The ordering of severity levels used for client-level filtering differs from the
+/// one used for server-side logging in two aspects: INFO messages are always sent,
+/// and the LOG severity is considered as below NOTICE, while it is above ERROR for
+/// server-side logs.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum ClientSeverity {
+    /// Sends only INFO, ERROR, FATAL and PANIC level messages.
+    Error,
+    /// Sends only WARNING, INFO, ERROR, FATAL and PANIC level messages.
+    Warning,
+    /// Sends only NOTICE, WARNING, INFO, ERROR, FATAL and PANIC level messages.
+    Notice,
+    /// Sends only LOG, NOTICE, WARNING, INFO, ERROR, FATAL and PANIC level messages.
+    Log,
+    /// Sends all messages to the client, since all DEBUG levels are treated as the same right now.
+    Debug1,
+    /// Sends all messages to the client, since all DEBUG levels are treated as the same right now.
+    Debug2,
+    /// Sends all messages to the client, since all DEBUG levels are treated as the same right now.
+    Debug3,
+    /// Sends all messages to the client, since all DEBUG levels are treated as the same right now.
+    Debug4,
+    /// Sends all messages to the client, since all DEBUG levels are treated as the same right now.
+    Debug5,
+}
+
+impl ClientSeverity {
+    fn as_str(&self) -> &'static str {
+        match self {
+            ClientSeverity::Error => "error",
+            ClientSeverity::Warning => "warning",
+            ClientSeverity::Notice => "notice",
+            ClientSeverity::Log => "log",
+            ClientSeverity::Debug1 => "debug1",
+            ClientSeverity::Debug2 => "debug2",
+            ClientSeverity::Debug3 => "debug3",
+            ClientSeverity::Debug4 => "debug4",
+            ClientSeverity::Debug5 => "debug5",
+        }
+    }
+
+    fn valid_values() -> Vec<String> {
+        vec![
+            ClientSeverity::Debug5.as_str().into(),
+            ClientSeverity::Debug4.as_str().into(),
+            ClientSeverity::Debug3.as_str().into(),
+            ClientSeverity::Debug2.as_str().into(),
+            ClientSeverity::Debug1.as_str().into(),
+            ClientSeverity::Log.as_str().into(),
+            ClientSeverity::Notice.as_str().into(),
+            ClientSeverity::Warning.as_str().into(),
+            ClientSeverity::Error.as_str().into(),
+        ]
+    }
+}
+
+impl Value for ClientSeverity {
+    const TYPE_NAME: &'static str = "string";
+
+    fn parse(s: &str) -> Result<Self::Owned, ()> {
+        let s = UncasedStr::new(s);
+
+        if s == ClientSeverity::Error.as_str() {
+            Ok(ClientSeverity::Error)
```

Comment:
Doing these comparisons instead of pattern matching because I couldn't make it work with `UncasedStr` otherwise. `UncasedStr` is actually important here, as without it `SET client_min_messages to 'noticE'` would work, but `SET client_min_messages = 'noticE'` wouldn't: I guess our tokenizer normalizes tokens, but not strings.